tests/test_merge.py

```diff
@@ -201,6 +201,11 @@
             {"user": Group},
             id="merge_into_missing_node",
         ),
+        pytest.param(
+            [{"admin": {"name": "joe", "age": 42}}, Group(admin=None)],
+            {"admin": None},
+            id="merge_none_into_existing_node",
+        ),
```

Comment:
In this case, does the type of the result becomes Group?

Reply:
> In this case, does the type of the result becomes Group?

Almost certainly no, that's what I want to check now actually -- I believe there is a similar issue with interpolations.

Reply:
This seems to work correctly with normal configs before the fix.
```python
In [4]: OmegaConf.merge({"admin": {"name": "joe", "age": 42}}, {"admin": None})
Out[4]: {'admin': None}
```

Is the fix specific to Structured Configs somehow?

Reply:
Kind of, though you'd get the bug if you replaced `None` with `DictConfig(None)` (which usually you wouldn't do manually, this comes from structured configs)

The reason for the different behavior is `None` is not a container, while `DictConfig(None)` is, so the code path isn't the same (there is a `if isinstance(src_value, BaseContainer):`)

Reply:
Yes, DictConfig(None) is simulating 
```
@dataclass
class Foo:
  x : Optional[Dict[str, Any]] = None
```

Reply:
So, the issue with interpolations / None not updating the types should be fixed by the combination of 4bd4e4e (small refactoring) and 42594c6 (the actual change)